cloud/services/scalesets/vmss.go

```diff
@@ -223,7 +221,41 @@ func (s *Service) Reconcile(ctx context.Context) error {
 			return errors.Wrapf(err, "cannot create VMSS")
 		}
 		s.Scope.V(2).Info("successfully created VMSS", "scale set", vmssSpec.Name)
+		s.Scope.SaveK8sVersion()
+	}
+
+	// get the VMSS to update status
+	existingVMSS, err := s.getExisting(ctx, vmssSpec.Name)
+	if err != nil {
+		return errors.Wrapf(err, "failed to get VMSS %s after create or update", vmssSpec.Name)
+	}
+
+	// check to see if we are running the most K8s version specified in the MachinePool spec
+	// if not, then update the instances that are not running that model
```

Comment:
Should we add a comment here about this being a temporary solution and that the end goal is to do rolling upgrade?

Reply:
How about I make an issue?

Reply:
Or maybe add your conclusions + current status to https://github.com/kubernetes-sigs/cluster-api-provider-azure/issues/819? that way we keep conversation history in one place